File: Partition_Array_According_to_Given_Pivot.py
nums = [9,12,5,10,14,3,10]
pivot = 10
def pivotArray(nums,pivot):
    ls = []
    ls2 = []
    mid = []
    for i in nums:
        if i < pivot:
            ls.append(i)
        elif i > pivot:
            ls2.append(i)
        else:
            mid.append(i)
    return ls + mid +ls2

print(pivotArray(nums,pivot))

File: Partition_Array_According_to_Given_Pivot/Partition_Array_According_to_Given_Pivot.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{9, 12, 5, 10, 14, 3, 10}
	pivot := 10
	fmt.Println(pivotArray(nums, pivot))
}

func pivotArray(nums []int, pivot int) []int {
	ls := []int{}
	ls2 := []int{}
	mid := []int{}
	mk := []int{}
	for _, i := range nums {
		if i < pivot {
			ls = append(ls, i)
		} else if i > pivot {
			ls2 = append(ls2, i)
		} else {
			mid = append(mid, i)
		}
	}
	mk = append(mk, ls...)
	mk = append(mk, mid...)
	mk = append(mk, ls2...)

	return mk
}
